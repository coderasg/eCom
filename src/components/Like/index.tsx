import React from 'react';

import  LikeIcon from './../Icons/LikeIcon/index.svg';

import styles from './like.module.scss';

type Props = {
  count: number;
};

function Like(props:Props) {
  const { count } = props;
  return (
    <span data-testid="Like-wrapper" className={styles.like}>
      <LikeIcon width={15} height={15} />
      <span className={styles.count}>{`${count}`}</span>
    </span>);
}

export default Like;
