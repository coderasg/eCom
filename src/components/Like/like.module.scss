.like {
  display: flex;
  align-items: center;

  .count {
    margin-left: 5px;
  }
}
