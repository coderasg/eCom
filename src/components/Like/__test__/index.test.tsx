import React from 'react';
import renderer from 'react-test-renderer';

import Like from '../index';

describe('(Component) Like', () => {
  const defaultProps = {
    count: 12
  };
  
  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<Like {...defaultProps} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
