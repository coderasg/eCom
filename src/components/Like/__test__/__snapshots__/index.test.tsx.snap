// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Like should pass snapshot test 1`] = `
<span
  data-testid="Like-wrapper"
>
  <SvgIcon
    height={15}
    width={15}
  />
  <span>
    12
  </span>
</span>
`;
